test/jasmine/tests/legend_test.js

```diff
@@ -628,3 +629,188 @@ describe('legend restyle update', function() {
+                expect(+legendItem.parentNode.style.opacity).toBeLessThan(1);
+            });
+            it('should unhide series', function(done) {
+                legendItem.dispatchEvent(new MouseEvent('mousedown'));
+                legendItem.dispatchEvent(new MouseEvent('mouseup'));
+                setTimeout(function() {
+                    expect(gd.data[2].visible).toBe(true);
+                    done();
+                }, DBLCLICKDELAY * 2);
+            });
+            it('should unfade legend item', function() {
+                expect(+legendItem.parentNode.style.opacity).toBe(1);
+            });
+        });
+        describe('double click', function() {
+            it('should hide series', function(done) {
+                legendItem.dispatchEvent(new MouseEvent('mousedown'));
+                legendItem.dispatchEvent(new MouseEvent('mouseup'));
+                legendItem.dispatchEvent(new MouseEvent('mousedown'));
+                legendItem.dispatchEvent(new MouseEvent('mouseup'));
+                setTimeout(function() {
+                    for(var i = 0; i < legendItems.length; i++) {
+                        if(i === testEntry) {
+                            expect(gd.data[i].visible).toBe(true);
+                        } else {
+                            expect(gd.data[i].visible).toBe('legendonly');
+                        }
+                    }
+                    done();
+                }, DBLCLICKDELAY);
+            });
+            it('should fade legend item', function() {
+                var legendItemi;
+                for(var i = 0; i < legendItems.length; i++) {
+                    legendItemi = legendItems[i];
+                    if(i === testEntry) {
+                        expect(+legendItemi.parentNode.style.opacity).toBe(1);
+                    } else {
+                        expect(+legendItemi.parentNode.style.opacity).toBeLessThan(1);
+                    }
+                }
+            });
+            it('should unhide series', function(done) {
+                legendItem.dispatchEvent(new MouseEvent('mousedown'));
+                legendItem.dispatchEvent(new MouseEvent('mouseup'));
+                legendItem.dispatchEvent(new MouseEvent('mousedown'));
+                legendItem.dispatchEvent(new MouseEvent('mouseup'));
+                setTimeout(function() {
+                    for(var i = 0; i < legendItems.length; i++) {
+                        expect(gd.data[i].visible).toBe(true);
+                    }
+                    done();
+                }, DBLCLICKDELAY);
+            });
+            it('should unfade legend items', function() {
+                var legendItemi;
+                for(var i = 0; i < legendItems.length; i++) {
+                    legendItemi = legendItems[i];
+                    expect(+legendItemi.parentNode.style.opacity).toBe(1);
+                }
+            });
+        });
+    });
+});
```

Comment:
Lowered the delays which uncovered a bug in the implementation, so both have been fixed.

Updated to `DBLCLICKDELAY` done in the `double_click` asset.

Didn't introduce any new click helpers for now, because those existing assets will be cause for confusion. Probably worth a general refactor as there are a few dozen mousedown+mouseup combo clicks across other tests.